ci-operator/step-registry/ipi/conf/aws/rootvolume/ipi-conf-aws-rootvolume-commands.sh

```diff
@@ -6,47 +6,159 @@ set -o pipefail
+  if [[ "${pool_name}" == "edge" ]]; then
+    echo "compute[1]"
+  else
+    echo ""
+  fi
+}
+
+echo "-------------------------------------------------------------"
+echo "Root volume configuration"
+echo "-------------------------------------------------------------"
+
+# Handle compute rootVolume configuration
+if [[ -n "${AWS_COMPUTE_VOLUME_TYPE:-}" ]] || [[ -n "${AWS_COMPUTE_GP3_THROUGHPUT:-}" ]]; then
+  configure_root_volume \
+    "compute[0]" \
+    "${AWS_COMPUTE_VOLUME_TYPE:-}" \
+    "${AWS_COMPUTE_VOLUME_SIZE:-}" \
+    "${AWS_COMPUTE_GP3_THROUGHPUT:-}" \
+    "compute"
 fi
 
-if [[ "${AWS_CONTROL_PLANE_VOLUME_TYPE}" != "" ]]; then
-  echo "Control plane volume type: ${AWS_CONTROL_PLANE_VOLUME_TYPE}"
-  PATCH=$(mktemp)
-  cat >> "${PATCH}" << EOF
-controlPlane:
-  platform:
-    aws:
-      rootVolume:
-        type: ${AWS_CONTROL_PLANE_VOLUME_TYPE}
-        size: ${AWS_CONTROL_PLANE_VOLUME_SIZE}
-EOF
-  cat "${PATCH}"
-  yq-go m -x -i "${CONFIG}" "${PATCH}"
+# Handle edge compute pool rootVolume configuration
+if [[ "${ENABLE_AWS_EDGE_ZONE:-}" == "yes" ]]; then
+  if [[ -n "${AWS_EDGE_VOLUME_TYPE:-}" ]] || [[ -n "${AWS_EDGE_GP3_THROUGHPUT:-}" ]]; then
+    edge_pool_path=$(get_edge_pool_path)
+    configure_root_volume \
+      "${edge_pool_path}" \
+      "${AWS_EDGE_VOLUME_TYPE:-}" \
+      "${AWS_EDGE_VOLUME_SIZE:-}" \
+      "${AWS_EDGE_GP3_THROUGHPUT:-}" \
+      "edge"
+    
+    # Output updated edge compute pool configuration
+    echo "Updated edge compute pool rootVolume configuration:"
+    yq-go r "${CONFIG}" "${edge_pool_path}.platform.aws.rootVolume" || true
+  fi
+fi
+
+# Handle controlPlane rootVolume configuration
+if [[ -n "${AWS_CONTROL_PLANE_VOLUME_TYPE:-}" ]] || [[ -n "${AWS_CONTROL_PLANE_GP3_THROUGHPUT:-}" ]]; then
+  configure_root_volume \
+    "controlPlane" \
+    "${AWS_CONTROL_PLANE_VOLUME_TYPE:-}" \
+    "${AWS_CONTROL_PLANE_VOLUME_SIZE:-}" \
+    "${AWS_CONTROL_PLANE_GP3_THROUGHPUT:-}" \
+    "controlPlane"
 fi
 
-if [[ "${AWS_DEFAULT_MACHINE_VOLUME_TYPE}" != "" ]]; then
-  echo "Default machine volume type: ${AWS_DEFAULT_MACHINE_VOLUME_TYPE}"
+# Handle defaultMachinePlatform rootVolume configuration
+# Note: defaultMachinePlatform applies to all pools unless overridden by specific pool settings
+if [[ "${AWS_DEFAULT_GP3_THROUGHPUT:-}" != "" ]]; then
   PATCH=$(mktemp)
   cat >> "${PATCH}" << EOF
 platform:
   aws:
     defaultMachinePlatform:
       rootVolume:
-        type: ${AWS_DEFAULT_MACHINE_VOLUME_TYPE}
-        size: ${AWS_DEFAULT_MACHINE_VOLUME_SIZE}
+        throughput: ${AWS_DEFAULT_GP3_THROUGHPUT}
+        iops: $(get_iops_from_throughput "${AWS_DEFAULT_GP3_THROUGHPUT}")
 EOF
   cat "${PATCH}"
   yq-go m -x -i "${CONFIG}" "${PATCH}"
 fi
+
+if [[ "${AWS_DEFAULT_MACHINE_VOLUME_TYPE:-}" != "" ]]; then
+  configure_root_volume \
+    "platform.aws.defaultMachinePlatform" \
+    "${AWS_DEFAULT_MACHINE_VOLUME_TYPE:-}" \
+    "${AWS_DEFAULT_MACHINE_VOLUME_SIZE:-}" \
+    "" \
```

Comment:
Ok, so this is the type and size for the default platform, why "" is passed here?